src/main/java/com/ericsson/gerrit/plugins/eiffel/listeners/AbstractEventListener.java

```diff
@@ -0,0 +1,131 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.ericsson.gerrit.plugins.eiffel.listeners;
+
+import java.io.File;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.gerrit.plugins.eiffel.configuration.EiffelPluginConfiguration;
+import com.google.gerrit.common.EventListener;
+import com.google.gerrit.reviewdb.client.Project;
+import com.google.gerrit.server.config.CanonicalWebUrl;
+import com.google.gerrit.server.events.ChangeEvent;
+import com.google.gerrit.server.events.Event;
+import com.google.inject.Inject;
+
+/**
+ * Abstract class implemented by the Gerrit event listeners, enforces needed listener methods and
+ * contains some helper methods to determine if Eiffel event sending is activated for the project
+ * the Gerrit event was sent by.
+ *
+ */
+public abstract class AbstractEventListener implements EventListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractEventListener.class);
+
+    @Inject
+    private com.google.gerrit.server.config.PluginConfigFactory pluginConfigFactory;
+
+    @Inject
+    @CanonicalWebUrl
+    private String gerritUrl;
+
+    protected final String pluginName;
+    protected final File pluginDir;
+
+    public AbstractEventListener(final String pluginName, final File pluginDir) {
+        this.pluginName = pluginName;
+        this.pluginDir = pluginDir;
+    }
+
+    /**
+     * Based on project name from the gerrit event, creates a project specific
+     * EiffelPluginConfiguration.
+     *
+     * @param gerritEvent
+     * @return EiffelPluginConfiguration
+     */
+    public EiffelPluginConfiguration createPluginConfig(final Event gerritEvent) {
+        ChangeEvent changeEvent = (ChangeEvent) gerritEvent;
+        final Project.NameKey projectNameKey = changeEvent.getProjectNameKey();
+        final EiffelPluginConfiguration pluginConfig = new EiffelPluginConfiguration(pluginName,
+                projectNameKey, pluginConfigFactory);
+        pluginConfig.setPluginDir(pluginDir);
+        return pluginConfig;
+    }
+
+    /**
+     * Returns a boolean indicating if Eiffel event sending is enabled for this project and branch.
+     *
+     * @param gerritEvent
+     * @param pluginConfig
+     * @return boolean
+     */
+    public boolean isEiffelEventSendingEnabled(final Event gerritEvent,
+            final EiffelPluginConfiguration pluginConfig) {
+
+        final ChangeEvent changeEvent = (ChangeEvent) gerritEvent;
+        final String project = changeEvent.change.get().project;
+        if (!isPluginEnabled(pluginConfig, project)) {
+            return false;
+        }
+
+        final String branch = changeEvent.change.get().branch;
+        final String filter = pluginConfig.getFilter();
+        boolean eiffelEventsShouldBeSent = isBranchNameInProjectFilterConfig(branch, filter,
```

Comment:
I would name this almost what the method is called, isEiffelEventSendingEnabled. Makes a bit of confusion

Reply:
Better?